twingate/datasource_groups.go

```diff
@@ -22,29 +29,73 @@ func datasourceGroupsRead(ctx context.Context, resourceData *schema.ResourceData
 	data := make([]interface{}, 0, len(groups))
 	for _, group := range groups {
 		data = append(data, map[string]interface{}{
-			"id":   group.ID.(string),
-			"name": string(group.Name),
+			"id":        group.ID.(string),
+			"name":      string(group.Name),
+			"type":      string(group.Type),
+			"is_active": bool(group.IsActive),
 		})
 	}
 
 	if err := resourceData.Set("groups", data); err != nil {
 		return diag.FromErr(err)
 	}
 
-	resourceData.SetId(groupName)
+	resourceData.SetId("groups-by-filter")
 
 	return diags
 }
 
+func buildFilter(resourceData *schema.ResourceData) *GroupsFilter {
+	groupName, hasName := resourceData.GetOk("name")
+	groupType, hasType := resourceData.GetOk("type")
+
+	// GetOk does not provide correct value for exists flag (second output value)
+	groupIsActive, hasIsActive := resourceData.GetOkExists("is_active") // nolint:staticcheck
+
+	if !hasName && !hasType && !hasIsActive {
+		return nil
```

Comment:
How is this not causing problems when there is no filtering and you pull all groups?

Reply:
I'm handling nil case in the GroupsFilter methods